### PR TITLE
Added r-constructive as a package

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thewchan** [comment]
@conda-forge/staged-recipes @conda-forge/help-r This is ready for review; though win build fails due to the same issues with other package. The package is no arch though

File: recipes/r-constructive/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  merge_build_host: True  # [win]
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
+requirements:
+  build:
+    - {{ compiler('c') }}              # [not win]
+    - {{ compiler('m2w64_c') }}        # [win]
```
Comment (mfansler) [inline recipes/r-constructive/meta.yaml:25]: Add stdlib's:

```suggestion
    - {{ compiler('c') }}        # [not win]
    - {{ stdlib('c') }}          # [not win]
    - {{ compiler('m2w64_c') }}  # [not win]
    - {{ stdlib('m2w64_c') }}    # [not win]
```

File: recipes/r-constructive/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    - {{ cran_mirror }}/src/contrib/Archive/constructive/constructive_{{ version }}.tar.gz
+  sha256: 307a540d91ec486c08f1e1dc82a228c54a6350d07ea635f3252a917a61f3e5f0
+
+build:
+  merge_build_host: True  # [win]
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
```
Comment (mfansler) [inline recipes/r-constructive/meta.yaml:21]: Missing DSO's:

```suggestion
  missing_dso_whitelist:
    - '*/R.dll'        # [win]
    - '*/Rblas.dll'    # [win]
    - '*/Rlapack.dll'  # [win]

```

Note that the latest version of `conda_r_skeleton_helper` includes this by default - be sure to update your copy!

File: recipes/r-constructive/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    - r-rlang >=1.0.0
+    - r-waldo
+
+test:
+  commands:
+    - $R -e "library('constructive')"           # [not win]
+    - "\"%R%\" -e \"library('constructive')\""  # [win]
+
+about:
+  home: https://github.com/cynkra/constructive, https://cynkra.github.io/constructive/
```
Comment (mfansler) [inline recipes/r-constructive/meta.yaml:52]: Split:

```suggestion
  home: https://cynkra.github.io/constructive/
  dev_url: https://github.com/cynkra/constructive
```

**Review by mfansler — CHANGES_REQUESTED** [review]
Thanks for submitting! ❤️ 

A few things to touch up then should be good.

**Comment by thewchan** [comment]
thanks @mfansler! All good to go now I think

**Review by mfansler — APPROVED** [review]
Great! (Sorry about my selector typo 😳)

**Pull request merged**